server/signal.go

```diff
@@ -28,7 +35,104 @@ func (s *Server) handleSignals() {
 			case syscall.SIGUSR1:
 				// File log re-open for rotating file logs.
 				s.ReOpenLogFile()
+			case syscall.SIGHUP:
+				// Config reload.
+				if err := s.Reload(); err != nil {
+					s.Errorf("Failed to reload server configuration: %s", err)
+				}
 			}
 		}
 	}()
 }
+
+// ProcessSignal sends the given signal command to the given process. If pidStr
+// is empty, this will send the signal to the single running instance of
+// gnatsd. If multiple instances are running, it returns an error. This returns
+// an error if the given process is not running or the command is invalid.
+func ProcessSignal(command Command, pidStr string) (err error) {
+	var pid int
+	if pidStr == "" {
+		pids, err := resolvePids()
```

Comment:
That I don't like. You use a named out-param `err` but here create a local `err` and the return the local `err` as the out parameter. Either declare pids first so that `err` is the global one or don't use a named parameter.